### PR TITLE
Document the new MinDomainsInPodTopologySpread feature gate

### PR DESCRIPTION
**Pull request by tengqm**
xref: kubernetes/kubernetes#107674

### CODE REVIEW

**Comment by mehabhalodiya** [comment]
enhancement ref: [kubernetes/enhancements/issues/3022](https://github.com/kubernetes/enhancements/issues/3022)

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 1]
```diff
@@ -73,16 +73,34 @@ You can define one or multiple `topologySpreadConstraint` to instruct the kube-s
...
+
 - **topologyKey** is the key of node labels. If two Nodes are labelled with this key and have identical values for that label, the scheduler treats both Nodes as being in the same topology. The scheduler tries to place a balanced number of Pods into each topology domain.
+
 - **whenUnsatisfiable** indicates how to deal with a Pod if it doesn't satisfy the spread constraint:
   - `DoNotSchedule` (default) tells the scheduler not to schedule it.
   - `ScheduleAnyway` tells the scheduler to still schedule it while prioritizing nodes that minimize the skew.
+
 - **labelSelector** is used to find matching Pods. Pods that match this label selector are counted to determine the number of Pods in their corresponding topology domain. See [Label Selectors](/docs/concepts/overview/working-with-objects/labels/#label-selectors) for more details.
 
 When a Pod defines more than one `topologySpreadConstraint`, those constraints are ANDed: The kube-scheduler looks for a node for the incoming Pod that satisfies all the constraints.
```
Comment (sanposhiho) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:106]: Could you add the example to describe minDomains?

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 2]
```diff
@@ -73,16 +73,34 @@ You can define one or multiple `topologySpreadConstraint` to instruct the kube-s
...
+    For example, if you have 3 zones with 0, 2 and 3 matching pods respectively,
+    The global minimum is 0).
   - when `whenUnsatisfiable` equals to "ScheduleAnyway", scheduler gives higher
     precedence to topologies that would help reduce the skew.
+
+- **minDomains** indicates a minimum number of eligible domains.
+  When the number of eligible domains with match topology keys is less than `minDomains`,
+  Pod topology spread treats "global minimum" as 0, and then the calculation of `skew` is performed.
+  When the number of eligible domains with matching topology keys equals or is greater than 
+  `minDomains`, this value has no effect on scheduling.
```
Comment (sanposhiho) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:90]: Add When `minDomains` is not nil, `whenUnsatisfiable` must be `DoNotSchedule.`

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 3]
```diff
@@ -73,16 +73,34 @@ You can define one or multiple `topologySpreadConstraint` to instruct the kube-s
...
     topology and the global minimum
-    (the minimum number of pods that match the label selector in a topology domain. For example, if you have 3 zones with 0, 2 and 3 matching pods respectively, The global minimum is 0).
+    (the minimum number of pods that match the label selector in a topology domain.
+    For example, if you have 3 zones with 0, 2 and 3 matching pods respectively,
+    The global minimum is 0).
   - when `whenUnsatisfiable` equals to "ScheduleAnyway", scheduler gives higher
     precedence to topologies that would help reduce the skew.
+
+- **minDomains** indicates a minimum number of eligible domains.
+  When the number of eligible domains with match topology keys is less than `minDomains`,
```
Comment (sanposhiho) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:87]: Describe what "eligible domains" is in somewhere.
https://github.com/kubernetes/kubernetes/blob/475f7af1c1ba17dd3787c609b8202696629b68c6/pkg/apis/core/types.go#L5629-L5630

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 4]
```diff
@@ -73,16 +73,34 @@ You can define one or multiple `topologySpreadConstraint` to instruct the kube-s
...
+  Pod topology spread treats "global minimum" as 0, and then the calculation of `skew` is performed.
+  When the number of eligible domains with matching topology keys equals or is greater than 
+  `minDomains`, this value has no effect on scheduling.
+
```
Comment (sanposhiho) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:91]: Add a description of the behavior in the case of `nil`.
https://github.com/kubernetes/kubernetes/blob/475f7af1c1ba17dd3787c609b8202696629b68c6/pkg/apis/core/types.go#L5669

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 5]
```diff
@@ -73,16 +73,34 @@ You can define one or multiple `topologySpreadConstraint` to instruct the kube-s
...
-    (the minimum number of pods that match the label selector in a topology domain. For example, if you have 3 zones with 0, 2 and 3 matching pods respectively, The global minimum is 0).
+    (the minimum number of pods that match the label selector in a topology domain.
+    For example, if you have 3 zones with 0, 2 and 3 matching pods respectively,
+    The global minimum is 0).
   - when `whenUnsatisfiable` equals to "ScheduleAnyway", scheduler gives higher
     precedence to topologies that would help reduce the skew.
+
+- **minDomains** indicates a minimum number of eligible domains.
+  When the number of eligible domains with match topology keys is less than `minDomains`,
+  Pod topology spread treats "global minimum" as 0, and then the calculation of `skew` is performed.
```
Comment (sanposhiho) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:88]: Describe what "global minimum" is in somewhere. 
https://github.com/kubernetes/kubernetes/blob/475f7af1c1ba17dd3787c609b8202696629b68c6/pkg/apis/core/types.go#L5607

**Review by sanposhiho — COMMENTED** [review]
Thanks for opening the PR instead of me 👍

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:106]: I'd prefer adding the example as a mermaid diagram in a separate PR.

**Comment by reylejano** [comment]
Code PR merged in https://github.com/kubernetes/kubernetes/pull/107674
/lgtm

**Comment by nate-double-u** [comment]
@PI-Victor, the LGTM got dropped after the last commit. Re-add and I'd be happy to approve.

**Review by chrisnegus — COMMENTED** [review]
/lgtm

**Pull request merged**